kubevirt-latency-checkup: Load parameters from environment

ormergi opened the pull request:
This PR  introduce the logic for loading and parsing Kubevirt VM Latency checkup parameters
from the environment.

**Depends on #41**

orelmisan commented on checkups/kubevirt-vm-latency/latencycheck/internal/config/config.go:
```diff
@@ -0,0 +1,125 @@
+const (
+	ResultsConfigMapNamespaceEnvVarName     = "RESULT_CONFIGMAP_NAMESPACE"
+	ResultsConfigMapNameEnvVarName          = "RESULT_CONFIGMAP_NAME"
+	NetworkNamespaceEnvVarName              = "NAD_NAMESPACE"
```
`NAD_NAMESPACE` should be `NETWORK_ATTACHMENT_DEFINITION_NAMESPACE`

Please see:
https://github.com/kiagnose/kiagnose/pull/12/files#diff-bbe57debe1510ab9407eca2922d081a04d7e6babacc15ecaf1d4275400db7d6dR71

ormergi replied:
Done

orelmisan commented on checkups/kubevirt-vm-latency/latencycheck/internal/config/config.go:
```diff
@@ -0,0 +1,125 @@
+	ResultsConfigMapNamespaceEnvVarName     = "RESULT_CONFIGMAP_NAMESPACE"
+	ResultsConfigMapNameEnvVarName          = "RESULT_CONFIGMAP_NAME"
+	NetworkNamespaceEnvVarName              = "NAD_NAMESPACE"
+	NetworkNameEnvVarName                   = "NAD_NAME"
```
`NAD_NAME` should be `NETWORK_ATTACHMENT_DEFINITION_NAME`.

Please see:
https://github.com/kiagnose/kiagnose/pull/12/files#diff-bbe57debe1510ab9407eca2922d081a04d7e6babacc15ecaf1d4275400db7d6dR72

ormergi replied:
Done

orelmisan commented on checkups/kubevirt-vm-latency/latencycheck/internal/config/config.go:
```diff
@@ -0,0 +1,125 @@
+
+var (
+	ErrEnvNotValid                       = errors.New("environment is not valid")
+	ErrResultsConfigMapNameMissing       = fmt.Errorf("%s environment variable is missing", ResultsConfigMapNameEnvVarName)
+	ErrResultsConfigMapNameNotValid      = fmt.Errorf("%s environment variable is not valid", ResultsConfigMapNameEnvVarName)
+	ErrResultsConfigMapNamespaceMissing  = fmt.Errorf("%s environment variable is missing", ResultsConfigMapNamespaceEnvVarName)
+	ErrResultsConfigMapNamespaceNotValid = fmt.Errorf("%s environment variable is not valid", ResultsConfigMapNamespaceEnvVarName)
+	ErrNetworkNameMissing                = fmt.Errorf("%s environment variable is missing", NetworkNameEnvVarName)
+	ErrNetworkNameNotValid               = fmt.Errorf("%s environment variable is not valid", NetworkNameEnvVarName)
+	ErrNetworkNamespaceMissing           = fmt.Errorf("%s environment variable is missing", NetworkNamespaceEnvVarName)
+	ErrNetworkNamespaceNotValid          = fmt.Errorf("%s environment variable is not valid", NetworkNamespaceEnvVarName)
```
Please consider: 
1. Changing `%s` to `%q` to make the variable name quoted.
2. Changing `is not valid` to `is invalid`.

```go
ErrResultsConfigMapNameInvalid  = fmt.Errorf("%q environment variable is invalid", ResultsConfigMapNameEnvVarName)
```

ormergi replied:
Done

orelmisan replied:
@ormergi is it possible to please change the variable names as well? (please see my suggestion in the original comment)

ormergi replied:
Done

orelmisan reviewed (commented):
Thank you for the PR @ormergi.
Please see the inline comments.

ormergi commented:
Changes: addressed code review comments

ormergi commented:
@orelmisan @EdDev please take another look

orelmisan commented:
@ormergi the unit tests are [failing](https://github.com/kiagnose/kiagnose/runs/6246449283?check_suite_focus=true).

ormergi commented:
rebase

ormergi commented:
[Changes](https://github.com/kiagnose/kiagnose/compare/a810d0c2604a05b4ae3392c6fb77caca0f596b5d..9732f88900969d1e0ce6c602a219536004afe9ed): Fixed unit tests and renamed variables

ormergi commented:
@orelmisan please take another look

ormergi commented:
/close 

This PR changes were included at #41
